default/python/tests/AI/bonus_calculator_tests.py

```diff
@@ -0,0 +1,17 @@
+import pytest
+from freeorion_tools.bonus_calculation import Bonus
+
+
+@pytest.mark.parametrize(
+    ("bonus", "expected"),
+    (
+        (Bonus(True, 1, 5.5), 0),
+        (Bonus(True, 2, 5.5), 0),
+        (Bonus(True, 0, 5.5), 5.5),
+        (Bonus(False, 0, 5.5), 0),
+        (Bonus(False, 1, 5.5), 0),
+        (Bonus(False, 2, 5.5), 0),
+    ),
+)
+def test_get_bonus_for_stability_1(bonus, expected):
+    assert bonus.get_bonus(1) == expected
```

Comment:
```suggestion
    ("bonus", "stability", "expected"),
    (
        (Bonus(True, 1, 5.5), 1, 5.5),
        (Bonus(True, 2, 5.5), 1.9, 0),
        (Bonus(True, 0, 5.5), 0.1, 5.5),
        (Bonus(False, 0, 5.5), 0, 0),
        (Bonus(False, 1, 5.5), 0, 0),
        (Bonus(False, 2, 5.5), 0, 0),
    ),
)
def test_get_bonus_for_stability_1(bonus, stability, expected):
    assert bonus.get_bonus(stability) == expected
```
When I looked at the table I wondered how you got to the value until I saw the test hard-coded stability. Looks more natural for me that way.

Reply:
The problem with parameterized tests is when you get too many arguments, you will need to look into the code anyway.   So I tried to reduce the number of them, hardcoding some in the test and the test name.